azure-kusto-data/azure/kusto/data/request.py

```diff
@@ -312,19 +303,55 @@ def _execute(
         if self._auth_provider:
             request_headers["Authorization"] = self._auth_provider.acquire_authorization_header()
 
+        timeout = self._get_timeout(properties, default_timeout)
         response = requests.post(endpoint, headers=request_headers, json=request_payload, timeout=timeout)
 
         if response.status_code == 200:
             if get_raw_response:
                 return response.json()
 
             if endpoint.endswith("v2/rest/query"):
-                kusto_response = KustoResponseDataSetV2(response.json())
+                return KustoResponseDataSetV2(response.json())
             else:
-                kusto_response = KustoResponseDataSetV1(response.json())
-
-            if kusto_response.errors_count > 0 and not accept_partial_results:
-                raise KustoServiceError(kusto_response.get_exceptions(), response, kusto_response)
-            return kusto_response
+                return KustoResponseDataSetV1(response.json())
         else:
             raise KustoServiceError([response.json()], response)
+
+    def _get_timeout(self, properties, default):
+        if properties:
+            if properties.has_option(ClientRequestProperties.OptionServerTimeout):
+                return properties.get_option_value_or_default(ClientRequestProperties.OptionServerTimeout, default)
+            return None

```

Comment:
I don't think this is what you meant.. if `properties` exists, but not `has_option`, we should still revert to default.

I would just change the entire function to 

`properties.get_value(ClientRequestProperties.OptionServerTimeout,default) if properties else default`